src/components/device/ha-device-trigger-picker.ts

```diff
@@ -0,0 +1,148 @@
+import "@polymer/paper-input/paper-input";
+import "@polymer/paper-item/paper-item";
+import "@polymer/paper-item/paper-item-body";
+import "@polymer/paper-dropdown-menu/paper-dropdown-menu-light";
+import "@polymer/paper-listbox/paper-listbox";
+import memoizeOne from "memoize-one";
+import {
+  LitElement,
+  TemplateResult,
+  html,
+  css,
+  CSSResult,
+  property,
+} from "lit-element";
+import { HomeAssistant } from "../../types";
+import { fireEvent } from "../../common/dom/fire_event";
+import computeStateName from "../../common/entity/compute_state_name";
+import {
+  DeviceTrigger,
+  fetchDeviceTriggers,
+  triggersEqual,
+} from "../../data/device_automation";
+
+class HaDeviceTriggerPicker extends LitElement {
+  public hass?: HomeAssistant;
+  @property() public label?: string;
+  @property() public deviceId?: string;
+  @property() public triggers: DeviceTrigger[] = [];
+  @property() public trigger?: DeviceTrigger;
+  @property() public presetTrigger?: DeviceTrigger;
+  private noTrigger: DeviceTrigger = {
+    device_id: this.deviceId || "",
+    platform: "device",
+  };
+
+  private _sortedTriggers = memoizeOne((triggers?: DeviceTrigger[]) => {
+    return triggers || [];
+  });
+
+  protected render(): TemplateResult | void {
+    const noTriggers = this._sortedTriggers(this.triggers).length === 0;
+    return html`
+      <paper-dropdown-menu-light .label=${this.label} ?disabled=${noTriggers}>
+        <paper-listbox
+          slot="dropdown-content"
+          .selected=${this._trigger}
```

Comment:
This cannot work because attributes are strings, and you cannot compare against an object. I suggest you create a unique key, like `{device_id}|{platform}` and use that.

Reply:
I think that it should also be possible to just use array index in the triggers array, and `-1` being no trigger.

Reply:
It's not a string attribute, it's a property though?

I'm not sure what you mean by `-1` being no trigger, do you mean for [fallbackSelection](https://www.webcomponents.org/element/@polymer/iron-menu-behavior/behaviors/IronMenuBehavior#property-fallbackSelection) ?

Reply:
yeah meant fallback selection. 

The way paper-listbox knows if something is selected, is by looking at the `attr-for-selected` value and calling `getAttribute()`. Since you set trigger as a property, not an attribute, it can't be used to mark the right one as selected ?

Reply:
It is actually working OK!

From the (docs)[https://www.webcomponents.org/element/@polymer/paper-
listbox/elements/paper-listbox#property-attrForSelected]:

> If you want to use an attribute value _or property_ of an element for selected instead of the index, set this to the name of the attribute _or property_.

Anyhow, maybe the code would be cleaner by using indices instead, I'll give it one more go.

Reply:
Rewritten now, maybe a bit cleaner.

Reply:
perfect